backend/rules/attackpath/PrivateServerlessAdmin.go

```diff
@@ -101,3 +102,34 @@ func (PrivateServerlessAdmin) Execute(tx neo4j.Transaction) ([]types.Result, err
 	}
 	return results, nil
 }
+
+func (PrivateServerlessAdmin) ProduceRuleGraph(tx neo4j.Transaction, resourceId string) (types.GraphPathResult, error) {
+	var params = make(map[string]interface{})
+	params["InstanceId"] = resourceId
+	// We lambda isn't exposed, we want to pick up the relevant role to return.
+	records, err := tx.Run(
+		`MATCH (a:AWSAccount{inscope: true})-[:RESOURCE]->(lambda:AWSLambda{id: $InstanceId})
+		WITH lambda
+		OPTIONAL MATCH
+			adminRolePath=
+			(lambda)-[:STS_ASSUME_ROLE_ALLOW]->(role:AWSRole{is_admin: True})
+		WITH lambda, collect(adminRolePath) as adminRolePaths
+		WITH adminRolePaths AS paths
+		RETURN paths`,
+		params,
+	)
+	if err != nil {
+		return types.GraphPathResult{}, err
+	}
+
+	processedGraphResult, err := graphprocessing.ProcessGraphPathResult(records, "paths")
```

Comment:
If this is used for every rule, do this call outside of this function - try keeping only rule specific stuff